File: android/app/src/main/java/gg/strims/ppspp/rpc/CAPeerClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class CAPeerClient(filepath: String) : RPCClient(filepath) {

    suspend fun renew(
        arg: CAPeerRenewRequest = CAPeerRenewRequest()
    ): CAPeerRenewResponse =
        this.callUnary("CAPeer/Renew", arg)

}

File: ios/App/App/API/CAPeerClient.swift
// swift-format-ignore-file
//
//  CAPeerClient.swift
//  App
//
//  Copyright © 2020 MemeLabs. All rights reserved.
//

import Foundation
import PromiseKit

class CAPeerClient {
  private var client: RPCClient

  init(_ client: RPCClient) {
    self.client = client
  }

  public func renew(_ arg: PBCAPeerRenewRequest = PBCAPeerRenewRequest()) -> Promise<PBCAPeerRenewResponse> {
    return self.client.callUnary("CAPeer/Renew", arg)
  }
  
}
